Sources/ComposableArchitecture/Reducer/Reducers/Reduce.swift

```diff
@@ -21,6 +21,27 @@ public struct Reduce<State, Action>: ReducerProtocol {
     self.init(internal: reduce)
   }
 
+  /// Initializes a reducer with a `reduce` function and explicit `State` and `Action` types to
+  /// assist the compiler in type checking.
+  ///
+  /// > Specifying `State.self` and `Action.self` as parameter is equivalent to specifying the
+  /// > generics in `Reduce<State, Action>`.
+  ///
+  /// - Parameters:
+  ///   - stateType: The type of state. Assists the compiler in type checking the reduce closure.
+  ///     Specify when autocompletion fails to suggest properties on state.
+  ///   - actionType: The type of action. Assists the compiler in type checking the reduce closure.
+  ///     Specify when autocompletion fails to suggest action cases.
+  ///   - reduce: A function that is called when ``reduce(into:action:)`` is invoked.
+  @inlinable
+  public init(
+    into stateType: State.Type,
+    action actionType: Action.Type,
```

Comment:
Added an overload because specifying _both_ is required for Swift to type-check either in the body. 😕